openassessment/xblock/openassessmentblock.py

```diff
@@ -1333,6 +1355,23 @@ def publish_assessment_event(self, event_name, assessment, **kwargs):
             event_data
         )
 
+    @XBlock.json_handler
+    def reset_submission(self, data, suffix=''):  # pylint: disable=unused-argument
+        """
+        Reset the student's submission.
+
+        Args:
+            data (dict): Unused parameter. Defaults to {}.
+            suffix (str, optional): Unused parameter. Defaults to ''.
+
+        Returns:
+            dict: A dictionary indicating the success status, e.g. { 'success': True }
+        """
+        block_user = self.runtime.service(self, "user").get_current_user()
+        user = get_user_by_username_or_email(block_user.opt_attrs['edx-platform.username'])
+        reset_student_attempts(self.course_id, user, self.location, user, True)  # pylint: disable=no-member
+        return {"success": True}
```

Comment:
What would it happen if there is an error?

Reply:
You are right. I included a try-except to handle any type of error.